onnx/onnx.in.proto

```diff
@@ -246,6 +351,10 @@ message ModelProto {
 
   // Named metadata values; keys should be distinct.
   repeated StringStringEntryProto metadata_props = 14;
+
+  // Training-specific information. Sequentially executing all stored
+  // "TrainingInfoProto"s is one training iteration.
```

Comment:
Can we clarify the semantics by adding the following?
```
In particular, this defines two functionalities: an initialization-step and a training-iteration-step.
The semantics of the initialization-step is that all initializers (in ModelProto.graph and in all
TrainingInfoProto.algorithm) are first initialized as specified by the initializers in the graph,
and then by the initialization_bindings in every instance in training_info.
The semantics of the training-iteration-step is that all the initializers are updated as specified
by the update_bindings in every instance in training_info.
```
In particular, I have a question: is my understanding of the initialization above correct?
In the ambiguous case where multiple initialization values are specified for W (e.g., if
the graph specifies an initializer value for initializer W,
and one or more initialization_bindings also specify a value for W.)

Reply:
Yes, your clarification looks nice to me. For ambiguity, only one of `algorithm` and `initialization` can be executed in an execution of all `TrainingInfoProto`s. If the user choose to train the model, they needs to execute all `algorithm`s and ignore all `initialization`s. If the user choose to reset the model, all `algorithm`s will be ignored and only `initialization`s will be sequentially executed.

Reply:
I added your clarification to the protobuf files. Thanks.